client/src/pages/Checkout.js

```diff
@@ -29,6 +38,31 @@ const Checkout = () => {
     0
   );
 
+  const getClientSecret = async () => {
+    const { data } = await API.post("/create-payment-intent", {
+      orderedItems: shoppingCart.map((item) => ({
+        id: item.id,
+        quantity: item.quantity,
+      })),
```

Comment:
i don't think you can send an array object like this. it will need to be `JSON.stringify(array)` on front end.  and then `import json` and `json.loads(list)` on backend to convert back to a regular python list.   (see my code at top of PUT in `user_handler.py` for converting cuisines)